### PR TITLE
chore(package): switch to react-hot-loader

### PR DESCRIPTION
**Pull request by layershifter**
Rel #1895.

[Details](https://github.com/gaearon/babel-plugin-react-transform)

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -165,18 +166,21 @@ if (!__TEST__) {
...
   //
   test: /\.js$/,
   exclude: /node_modules/,
-  use: {
-    loader: 'babel-loader',
-    options: {
-      cacheDirectory: true,
-    },
-  },
+  loaders: jsLoaders,
```
Comment (levithomason) [inline webpack.config.js]: I think this should be a `use: []` and not `loaders: []`, correct?

https://webpack.js.org/concepts/loaders/#configuration

Comment (layershifter) [reply webpack.config.js]: @levithomason Thanks for your eyes, [docs](https://github.com/gaearon/react-hot-loader/tree/master/docs#migration-to-30) sent me the wrong way

File: webpack.config.js [hunk 2]
```diff
@@ -165,18 +166,21 @@ if (!__TEST__) {
 // ------------------------------------
 // Rules
 // ------------------------------------
+const jsLoaders = [{
+  loader: 'babel-loader',
+  options: {
+    cacheDirectory: true,
+  },
+}]
+if (__DEV__) jsLoaders.unshift('react-hot-loader/webpack')
```
Comment (levithomason) [inline webpack.config.js:175]: Does this work as is?  I'm not as solid with webpack 2 yet, but the [docs](https://webpack.js.org/concepts/loaders/#configuration) show only this syntax:

```diff
-if (__DEV__) jsLoaders.unshift('react-hot-loader/webpack')
+if (__DEV__) jsLoaders.unshift({ loader: 'react-hot-loader/webpack' })
```

Comment (layershifter) [reply webpack.config.js:175]: I can find this in docs, but it [works](http://andrewhfarmer.com/webpack-2-loader-variations/#3-rule-with-multiple-loaders-no-options).

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.71.5`

**Pull request merged**